### PR TITLE
Fix issue #124 - threading properties

### PR DESCRIPTION
**Pull request by markglh**
Fixes issue https://github.com/awslabs/amazon-kinesis-producer/issues/124

This is the quick fix imho. The long term fix for the property resolution is to migrate to the KCL approach I described in the issue. This would also then allow support for CredentialProviders too.

### CODE REVIEW

**Comment by markglh** [comment]
The formatting changes are in line with java standards - but I'm happy to undo them if it's an issue.

File: java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/KinesisProducerConfiguration.java [hunk 1]
```diff
@@ -1340,15 +1328,25 @@ public KinesisProducerConfiguration setThreadingModel(ThreadingModel threadingMo
...
     /**
-     * Sets the maximum number of threads that the native process' thread pool will be configured with.
+     * Sets the threading model that the native process will use.
+     * <p>
+     * See {@link #getThreadingModel()} for more information
      *
+     * @param threadingModel the String representation of the threading model to use.
+     * @return this configuration object
+     */
+    public KinesisProducerConfiguration setThreadingModel(String threadingModel) {
```
Comment (bowenli86) [inline java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/KinesisProducerConfiguration.java]: can you add valid strings to both this java doc and https://github.com/awslabs/amazon-kinesis-producer/blob/master/java/amazon-kinesis-producer-sample/default_config.properties for reference

File: java/amazon-kinesis-producer/src/test/java/com/amazonaws/services/kinesis/producer/KinesisProducerConfigurationTest.java [hunk 2]
```diff
@@ -79,12 +79,29 @@ public void loadBoolean() {
...
-    public void unknownProperty() { 
+    public void unknownProperty() {
         Properties p = new Properties();
         p.setProperty("xcdfndetnedtne5tje45", "Sfbsfrne34534");
         KinesisProducerConfiguration.fromPropertiesFile(writeFile(p));
         // should not throw exception
     }
+
+    @Test
+    public void setThreadingModelFromProperties() {
```
Comment (bowenli86) [inline java/amazon-kinesis-producer/src/test/java/com/amazonaws/services/kinesis/producer/KinesisProducerConfigurationTest.java]: can you also test PER_REQUEST?

**Comment by markglh** [comment]
Comments addressed @bowenli86 

I've also updated the `.gitignore` to prevent files from various IDEs being picked up

**Comment by bowenli86** [comment]
LGTM!

**Comment by markglh** [comment]
Thanks @bowenli86 - how do we get this one merged/released? :)

**Comment by bowenli86** [comment]
@markglh ask an AWS guy? I'm not from Amazon, so I don't have access

**Comment by markglh** [comment]
Ah I thought you might have been working with/for them!

I'm guessing @pfifer is the one to prod? :D

**Comment by pfifer** [comment]
Thanks for providing, but a small request:  Can you back out the whitespace changes to the KinesisProducerConfiguration.java?  It makes reviewing the diff more difficult.

Thanks

**Comment by markglh** [comment]
@pfifer no worries, that's all done now! Apologies for the delay, I've been on vacation :)

Cheers!

**Comment by pfifer** [comment]
Please confirm that we can use, modify, copy, and redistribute this contribution. Thanks.

**Comment by markglh** [comment]
@pfifer of course: `you can use, modify, copy, and redistribute this contribution.`

**Comment by bowenli86** [comment]
@pfifer can you take a look and release a new version after merging this PR?

**Comment by markglh** [comment]
Yeah 2 months to merge makes me sad :(

**Comment by bowenli86** [comment]
@markglh You've done a great job on this :)  I'm upset about Amazon's speed too.

**Comment by pfifer** [comment]
Sorry for the delay.  I will be merging this soon.

**Comment by bowenli86** [comment]
Hi @pfifer , when can Amzn release KPL 0.10.6 with this fix?

**Pull request merged**